firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -407,27 +419,37 @@ private static void logBadConfigError() {
             + " initializing Firebase.");
   }
 
-  private HttpURLConnection openHttpURLConnection(URL url, String apiKey) throws IOException {
-    HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection();
-    httpURLConnection.setConnectTimeout(NETWORK_TIMEOUT_MILLIS);
-    httpURLConnection.setUseCaches(false);
-    httpURLConnection.setReadTimeout(NETWORK_TIMEOUT_MILLIS);
-    httpURLConnection.addRequestProperty(CONTENT_TYPE_HEADER_KEY, JSON_CONTENT_TYPE);
-    httpURLConnection.addRequestProperty(ACCEPT_HEADER_KEY, JSON_CONTENT_TYPE);
-    httpURLConnection.addRequestProperty(CONTENT_ENCODING_HEADER_KEY, GZIP_CONTENT_ENCODING);
-    httpURLConnection.addRequestProperty(CACHE_CONTROL_HEADER_KEY, CACHE_CONTROL_DIRECTIVE);
-    httpURLConnection.addRequestProperty(X_ANDROID_PACKAGE_HEADER_KEY, context.getPackageName());
-    if (heartbeatInfo != null && userAgentPublisher != null) {
-      HeartBeat heartbeat = heartbeatInfo.getHeartBeatCode(FIREBASE_INSTALLATIONS_ID_HEARTBEAT_TAG);
-      if (heartbeat != HeartBeat.NONE) {
-        httpURLConnection.addRequestProperty(USER_AGENT_HEADER, userAgentPublisher.getUserAgent());
-        httpURLConnection.addRequestProperty(
-            HEART_BEAT_HEADER, Integer.toString(heartbeat.getCode()));
+  private HttpURLConnection openHttpURLConnection(URL url, String apiKey)
+      throws FirebaseInstallationsException {
+    try {
+      HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection();
```

Comment:
is it maybe enough to wrap this line with the try-catch block?
Maybe it gives more clarity about what can go wrong here.

Reply:
Done.